projects/observability/src/shared/components/gauge-list/gauge-list.component.ts

```diff
@@ -71,22 +74,35 @@ export class GaugeListComponent<T extends GaugeItem = GaugeItem> implements OnCh
       return;
     }
 
-    let maxValue = maxBy(this.items, option => option.value)?.value;
-    if (maxValue === undefined || maxValue === 0) {
-      maxValue = 1;
-    }
-
     const colorLookupMap = this.buildColorLookupForData(this.items);
-    const totalCount = sum(this.items.map((gaugeItem: GaugeItem) => gaugeItem.value));
-
-    this.itemOptions = this.items.map(option => ({
-      label: option.label,
-      color: colorLookupMap.get(option.colorKey ?? option.label),
-      width: `${((option.value / maxValue!) * 100).toFixed(2)}%`,
-      value: option.value,
-      percentage: totalCount > 0 ? (option.value / totalCount) * 100 : 100,
-      original: option
-    }));
+
+    if (this.total !== undefined && this.total !== 0) {
+      this.itemOptions = this.items.map(option => ({
+        label: option.label,
+        color: colorLookupMap.get(option.colorKey ?? option.label),
+        width: `${((option.value / this.total!) * 100).toFixed(2)}%`,
+        value: option.value,
+        percentage: (option.value / this.total!) * 100,
+        original: option
```

Comment:
This looks duplicate, same as L97 to L103

Reply:
can we create a function for those and pass the values as arguments

Reply:
Updated. Thanks!